javascript/javascript3/week3/shoppingCart/script.js

```diff
@@ -0,0 +1,174 @@
+  };
+
+  const windowListener = (e) => {
+    const parentTarget = e.target.parentElement;
+
+    if (parentTarget.classList.contains('product__img')) {
+      const productId = +parentTarget.dataset.img;
+
+      productToAdd = shoppingCart.renderProductInfo(productId, productPopup);
+      productPopup.classList.add('popup_active');
+      document.body.style.overflow = 'hidden';
+    } else if (parentTarget.classList.contains('search__item')) {
+      const productId = parentTarget.dataset.search;
+
+      productToAdd = productSearchDB.renderAndReturnProductPopup(
+        productId,
+        productPopup
+      );
+      productPopup.classList.add('popup_active');
+      closeSearchList();
+      document.body.style.overflow = 'hidden';
+    } else if (!parentTarget.classList.contains('search__item')) {
+      closeSearchList();
+    }
+  };
+
+  const popupListener = (e) => {
+    if (
+      (e.target.classList.contains('product__popup') &&
+        productPopup.classList.contains('popup_active')) ||
+      e.target.parentElement.classList.contains('btn_close')
+    ) {
+      productPopup.classList.remove('popup_active');
+      document.body.style.overflow = 'scroll';
+    } else if (e.target.classList.contains('btn_add')) {
+      const newProduct = new Product({ ...productToAdd, amount: 0 });
+      shoppingCart.addProduct(newProduct);
+      shoppingCart.renderProducts();
+      changeProductAmount(newProduct, 'increaseAmount');
+      productSearchDB.renderAddedProduct(newProduct, productPopup);
+      setTimeout(() => {
+        productPopup.classList.remove('popup_active');
+      }, 2000);
+      document.body.style.overflow = 'scroll';
+    }
+  };
+
+  const changeProductAmount = (productToChange, changingAction) => {
+    const amountInput = document.querySelector(
+      `[data-input="${productToChange.id}"]`
+    );
+    const priceOutput = document.querySelector(
+      `[data-price="${productToChange.id}"]`
+    );
+
+    const productId = productToChange.id;
+    const product = shoppingCart[changingAction](productId);
+    amountInput.value = product.amount;
+    priceOutput.innerHTML = `${parseFloat(
+      (product.amount * product.price).toFixed(1)
+    )} ${currentCurrency.currency}`;
+    totalPrice.innerHTML = `Total price: ${shoppingCart.getTotal()}`;
+  };
+
+  const shoppingListListener = (e) => {
+    const parentTarget = e.target.parentElement;
+
+    if (parentTarget.classList.contains('btn_delete')) {
+      shoppingCart.removeProduct(+parentTarget.dataset.id);
+      shoppingCart.renderProducts();
+      totalPrice.innerHTML = `Total price: ${shoppingCart.getTotal()}`;
+    } else if (parentTarget.classList.contains('arrow-left')) {
+      changeProductAmount(parentTarget.dataset, 'decreaseAmount');
+    } else if (parentTarget.classList.contains('arrow-right')) {
+      changeProductAmount(parentTarget.dataset, 'increaseAmount');
+    }
+  };
+
+  const changeCurrency = () => {
```

Comment:
This function is removing old products with a new shoppingCart list. But we are losing all amounts from products when we change currency. Maybe you can save amounts before changing currency. With some small changes you can use default amounts in your products with new currency. But this is just a suggestion. All other parts work pretty well.
fx.
`const amounts = shoppingCart.products.map(product => product.amount);`
`shoppingCart = new ShoppingCart(shoppingList, currentCurrency, amounts);`